src/transformers/generation/stopping_criteria.py

```diff
@@ -485,7 +487,8 @@ def __init__(self, eos_token_id: Union[int, List[int], torch.Tensor]):
     @add_start_docstrings(STOPPING_CRITERIA_INPUTS_DOCSTRING)
     def __call__(self, input_ids: torch.LongTensor, scores: torch.FloatTensor, **kwargs) -> torch.BoolTensor:
         self.eos_token_id = self.eos_token_id.to(input_ids.device)
-        if input_ids.device.type == "mps":
+        if input_ids.device.type == "mps" and not is_torch_greater_or_equal_than_2_4:
```

Comment:
Not sure we necessarily want to throw a warning to the user here @gante, since we already have a custom workaround for mps? Rather, we can leave it as a TODO to remove the workaround when we drop 2.3 support

Reply:
yeah, makes sense!